multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1065,35 +1153,66 @@ def _predict(
+        chunk_size: Optional[int] = 1024,
+        similarity_type: Optional[str] = "cosine",
+        top_k: Optional[int] = 100,
     ):
-        """
-        Evaluate model on a test dataset.
+        query_header = self._query[0] if self._query is not None else QUERY
+        query_data = data_to_df(data=query_data, header=query_header)
+        response_header = self._response[0] if self._response else RESPONSE
+        response_data = data_to_df(data=response_data, header=response_header)
 
-        Parameters
-        ----------
-        data
-            A dataframe, containing the same columns as the training data
-        id_mappings
-             Id-to-content mappings. The contents can be text, image, etc.
-             This is used when the dataframe contains the query/response indexes instead of their contents.
-        metrics
-            A list of metric names to report.
-            If None, we only return the score for the stored `_eval_metric_name`.
-        return_pred
-            Whether to return the prediction result of each row.
+        qr_relevance = data_to_df(data=qr_relevance)
 
-        Returns
-        -------
-        A dictionary with the metric names and their corresponding scores.
-        Optionally return a dataframe of prediction results.
-        """
+        rank_labels = {}
+        for i, per_row in qr_relevance.iterrows():
+            rank_labels.setdefault(per_row[self._query[0]], {})[per_row[self._response[0]]] = int(
+                per_row[self._label_column]
+            )
+
+        rank_results = dict()
+        query_embeddings = self.extract_embedding(query_data, signature=QUERY, id_mappings=id_mappings, as_tensor=True)
+        num_chunks = max(1, len(response_data) // chunk_size)
+        for response_chunk in np.array_split(response_data, num_chunks):
+            response_embeddings = self.extract_embedding(
+                response_chunk, signature=RESPONSE, id_mappings=id_mappings, as_tensor=True
+            )
+            similarity_scores = compute_semantic_similarity(
+                a=query_embeddings, b=response_embeddings, similarity_type=similarity_type
+            )
+            similarity_scores[torch.isnan(similarity_scores)] = -1
+            top_k_scores, top_k_indices = torch.topk(
+                similarity_scores,
+                k=min(top_k + 1, len(similarity_scores[1])),
+                dim=1,
+                largest=True,
+                sorted=False,
+            )
+            top_k_indices = top_k_indices.cpu().tolist()
+            top_k_scores = top_k_scores.cpu().tolist()
+            for i in range(len(query_data)):
+                query_idx = query_data.iloc[i][self._query[0]]
+                for sub_response_idx, score in zip(top_k_indices[i], top_k_scores[i]):
+                    response_idx = response_chunk.iloc[int(sub_response_idx)][self._response[0]]
+                    rank_results.setdefault(query_idx, {})[response_idx] = score
+
```

Comment:
consider other scores like MRR

Reply:
Added a TODO in `compute_ranking_score`.